presto-main/src/main/java/io/prestosql/server/DynamicFilterService.java

```diff
@@ -228,9 +229,7 @@ public void unblockStageDynamicFilters(QueryId queryId, PlanFragment plan)
     public DynamicFilter createDynamicFilter(QueryId queryId, List<DynamicFilters.Descriptor> dynamicFilterDescriptors, Map<Symbol, ColumnHandle> columnHandles)
     {
         Multimap<DynamicFilterId, ColumnHandle> sourceColumnHandles = extractSourceColumnHandles(dynamicFilterDescriptors, columnHandles);
-        Set<DynamicFilterId> dynamicFilters = dynamicFilterDescriptors.stream()
-                .map(DynamicFilters.Descriptor::getId)
-                .collect(toImmutableSet());
+        Set<DynamicFilterId> dynamicFilters = ImmutableSet.copyOf(sourceColumnHandles.keySet());
```

Comment:
nit: separate commit

Reply:
I would prefer to keep it within this commit as it's not an unrelated refactoring. This change is needed to avoid failing the checkState in `translateSummaryToTupleDomain`.